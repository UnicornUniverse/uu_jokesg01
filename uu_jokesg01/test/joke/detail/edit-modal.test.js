import { mount, wait } from "uu5g05-test";
import { Button, MenuItem } from "uu5g05-elements";
import Detail from "../../../src/joke/detail";
import EditModal from "../../../src/joke/detail/edit-modal";
import Uu5Editing from "uu5g05-editing";

describe(`UuJokes.Joke.Detail.EditModal`, () => {
  it(`checks props passed to children`, async () => {
    const props = {
      componentType: Detail,
      componentProps: { oid: "joke-1" },
      onSave: () => {},
      onClose: () => {},
      onReady: () => {},
    };

    const spy = jest.spyOn(Uu5Editing, "EditModal");
    const Fallback = jest.fn(() => <div>Loading...</div>);

    mount(<EditModal {...props} fallback={<Fallback />} />);
    await wait();

    expect(spy.mock.lastCall[0]).toMatchSnapshot();
    expect(Fallback).toHaveBeenCalledTimes(1);
  });

  it(`opens advanced settings`, async () => {
    const props = {
      componentType: Detail,
      componentProps: { oid: "joke-1" },
      onSave: () => {},
      onClose: () => {},
      onReady: () => {},
    };

    const Fallback = jest.fn(() => <div>Loading...</div>);
    const wrapper = mount(<EditModal {...props} fallback={<Fallback />} />);
    await wait();

    wrapper
      .find(MenuItem)
      .findWhere((node) => node.text() === "Advanced settings")
      .first()
      .simulate("click");
    await wait();

    expect(wrapper.contains("Heading size")).toBe(true);
  });

  it(`saves props`, async () => {
    const props = {
      componentType: Detail,
      componentProps: { oid: "joke-1", baseUri: "http://localhost" },
      onSave: jest.fn(),
      onClose: () => {},
      onReady: () => {},
    };

    const Fallback = jest.fn(() => <div>Loading...</div>);
    const wrapper = mount(<EditModal {...props} fallback={<Fallback />} />);
    await wait();

    wrapper
      .find(Button)
      .findWhere((node) => node.contains("Save"))
      .first()
      .simulate("click");
    await wait();

    expect(props.onSave).toHaveBeenCalledTimes(1);
  });
});
