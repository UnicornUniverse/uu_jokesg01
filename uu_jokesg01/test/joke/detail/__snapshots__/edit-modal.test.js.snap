// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UuJokes.Joke.Detail.EditModal checks props passed to children 1`] = `
Object {
  "header": <Uu5g05.Lsi
    lsi={
      Object {
        "cs": "Upravit Joke.Detail",
        "en": "Edit Joke.Detail",
      }
    }
    path={Array []}
  />,
  "onClose": [Function],
  "onSave": [Function],
  "open": true,
  "propInputMap": Object {
    "aspectRatio": Object {
      "props": Object {
        "valueList": Array [
          "1:1",
          "2:1",
          "2:3",
          "3:1",
          "3:2",
          "3:4",
          "4:3",
          "4:5",
          "5:4",
          "16:9",
          "16:10",
        ],
      },
    },
    "baseUri": Object {
      "props": Object {
        "label": Object {
          "cs": "URL aplikace uuJokes",
          "en": "uuJokes application URL",
        },
        "required": true,
      },
    },
    "disableUserPreference": Object {
      "component": [Function],
      "props": Object {
        "label": Object {
          "cs": "Zakázat uživatelské nastavení",
          "en": "Forbid user configuration",
        },
      },
    },
    "level": Object {
      "component": [Function],
      "name": "level",
      "props": [Function],
    },
    "nestingLevel": Object {
      "props": Object {
        "valueList": Array [
          "area",
          "box",
          "spot",
          "inline",
        ],
      },
    },
    "oid": Object {
      "component": [Function],
      "props": Object {
        "label": Object {
          "cs": "Identifikátor (oid)",
          "en": "Identifier (oid)",
        },
        "required": true,
      },
    },
    "showAuthor": Object {
      "component": [Function],
      "name": "showAuthor",
      "props": Object {
        "label": Object {
          "cs": "Zobrazit autora",
          "en": "Show author",
        },
      },
    },
    "showCategories": Object {
      "component": [Function],
      "name": "showCategories",
      "props": Object {
        "label": Object {
          "cs": "Zobrazit kategorie",
          "en": "Show categories",
        },
      },
    },
    "showCreationTime": Object {
      "component": [Function],
      "name": "showCreationTime",
      "props": Object {
        "label": Object {
          "cs": "Zobrazit čas vytvoření",
          "en": "Show creation time",
        },
      },
    },
  },
  "props": Object {
    "oid": "joke-1",
  },
  "tabList": Array [
    Object {
      "label": Object {
        "cs": "Vtip",
        "en": "Joke",
      },
      "layout": Object {
        "xs": "baseUri, oid",
      },
    },
    Object {
      "label": Object {
        "cs": "Výchozí nastavení",
        "en": "Default configuration",
      },
      "layout": Object {
        "xs": "showCategories, showAuthor, showCreationTime, .,disableUserPreference",
      },
    },
    Object {
      "columns": "1fr 1fr",
      "layout": Object {
        "xs": "
            identificationType identificationType,
            nestingLevel nestingLevel,
            card card,
            significance significance,
            borderRadius borderRadius,
            aspectRatio aspectRatio,
            width height
          ",
      },
      "template": "visual",
    },
    Object {
      "label": Object {
        "cs": "Pokročilá nastavení",
        "en": "Advanced settings",
      },
      "layout": Object {
        "xs": "level",
      },
    },
  ],
  "uu5Tag": "Plus4U5Elements.withErrorBoundary(withEditMode(withMargin(UuJokesCore.Joke.Detail)))",
}
`;
